File: web/src/api/modeltraining/trainingTask.js
import service from '@/utils/request'

/**
 * 训练任务管理 API
 */

// 创建训练任务
export const createTask = (data) => {
  return service({
    url: '/modeltraining/trainingTask/createTask',
    method: 'post',
    data
  })
}

// 删除训练任务
export const deleteTask = (params) => {
  return service({
    url: '/modeltraining/trainingTask/deleteTask',
    method: 'delete',
    params
  })
}

// 批量删除训练任务
export const deleteTaskByIds = (params) => {
  return service({
    url: '/modeltraining/trainingTask/deleteTaskByIds',
    method: 'delete',
    params
  })
}

// 更新训练任务
export const updateTask = (data) => {
  return service({
    url: '/modeltraining/trainingTask/updateTask',
    method: 'put',
    data
  })
}

// 查询训练任务详情
export const findTask = (params) => {
  return service({
    url: '/modeltraining/trainingTask/findTask',
    method: 'get',
    params
  })
}

// 分页获取训练任务列表
export const getTaskList = (params) => {
  return service({
    url: '/modeltraining/trainingTask/getTaskList',
    method: 'get',
    params
  })
}

// 启动训练任务
export const startTask = (params) => {
  return service({
    url: '/modeltraining/trainingTask/startTask',
    method: 'post',
    params
  })
}

// 停止训练任务
export const stopTask = (params) => {
  return service({
    url: '/modeltraining/trainingTask/stopTask',
    method: 'post',
    params
  })
}

// 获取训练日志
export const getTaskLogs = (params) => {
  return service({
    url: '/modeltraining/trainingTask/getTaskLogs',
    method: 'get',
    params
  })
}

// 获取训练任务数据源
export const getTaskDataSource = () => {
  return service({
    url: '/modeltraining/trainingTask/getTaskDataSource',
    method: 'get'
  })
}

// 获取默认训练参数
export const getDefaultParams = () => {
  return service({
    url: '/modeltraining/trainingTask/getDefaultParams',
    method: 'get'
  })
}

// 模型对话测试
export const chatCompletion = (data) => {
  return service({
    url: '/modeltraining/trainingTask/chatCompletion',
    method: 'post',
    data
  })
}

// 启动推理服务
export const startService = (params) => {
  return service({
    url: '/modeltraining/trainingTask/startService',
    method: 'post',
    params
  })
}

// 停止推理服务
export const stopService = (params) => {
  return service({
    url: '/modeltraining/trainingTask/stopService',
    method: 'post',
    params
  })
}

// 手动标记训练完成
export const markCompleted = (params) => {
  return service({
    url: '/modeltraining/trainingTask/markCompleted',
    method: 'post',
    params
  })
}

// ========== 模型测试历史 API ==========

// 创建测试历史
export const createTestHistory = (data) => {
  return service({
    url: '/modeltraining/modelTest/createTestHistory',
    method: 'post',
    data
  })
}

// 删除测试历史
export const deleteTestHistory = (params) => {
  return service({
    url: '/modeltraining/modelTest/deleteTestHistory',
    method: 'delete',
    params
  })
}

// 获取测试历史列表
export const getTestHistoryList = (params) => {
  return service({
    url: '/modeltraining/modelTest/getTestHistoryList',
    method: 'get',
    params
  })
}

// 清空测试历史
export const clearTestHistory = (params) => {
  return service({
    url: '/modeltraining/modelTest/clearTestHistory',
    method: 'delete',
    params
  })
}

File: server/router/modeltraining/training_task.go
package modeltraining

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type TrainingTaskRouter struct{}

// InitTrainingTaskRouter 初始化训练任务路由
func (r *TrainingTaskRouter) InitTrainingTaskRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	taskRouter := Router.Group("modeltraining/trainingTask").Use(middleware.OperationRecord())
	taskRouterWithoutRecord := Router.Group("modeltraining/trainingTask")
	taskRouterWithoutAuth := PublicRouter.Group("modeltraining/trainingTask")
	{
		taskRouter.POST("createTask", trainingTaskApi.CreateTask)             // 创建训练任务
		taskRouter.DELETE("deleteTask", trainingTaskApi.DeleteTask)           // 删除训练任务
		taskRouter.DELETE("deleteTaskByIds", trainingTaskApi.DeleteTaskByIds) // 批量删除训练任务
		taskRouter.PUT("updateTask", trainingTaskApi.UpdateTask)              // 更新训练任务
		taskRouter.POST("startTask", trainingTaskApi.StartTask)               // 启动训练
		taskRouter.POST("stopTask", trainingTaskApi.StopTask)                 // 停止训练
		taskRouter.POST("markCompleted", trainingTaskApi.MarkCompleted)       // 手动标记完成
		taskRouter.POST("startService", trainingTaskApi.StartService)         // 启动推理服务
		taskRouter.POST("stopService", trainingTaskApi.StopService)           // 停止推理服务
		taskRouter.POST("chatCompletion", trainingTaskApi.ChatCompletion)     // 模型对话测试
	}
	{
		taskRouterWithoutRecord.GET("findTask", trainingTaskApi.FindTask)                  // 查询任务详情
		taskRouterWithoutRecord.GET("getTaskList", trainingTaskApi.GetTaskList)            // 获取任务列表
		taskRouterWithoutRecord.GET("getTaskLogs", trainingTaskApi.GetTaskLogs)            // 获取训练日志
		taskRouterWithoutRecord.GET("getDefaultParams", trainingTaskApi.GetDefaultParams)  // 获取默认参数
	}
	{
		taskRouterWithoutAuth.GET("getTaskDataSource", trainingTaskApi.GetTaskDataSource) // 获取数据源
	}
}